src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/ModellingGroupController.kt

```diff
@@ -160,30 +139,40 @@ open class ModellingGroupController(context: ControllerContext)
             .filter { it.name == "modelling-groups.manage-members" }
             .map { it.scope }
 
-    open fun addBurdenEstimate(context: ActionContext, estimateRepository: BurdenEstimateRepository): String
+    fun addBurdenEstimatesFromHTMLForm(context: ActionContext, estimateRepository: BurdenEstimateRepository): String
     {
         // First check if we're allowed to see this touchstone
         val path = getValidResponsibilityPath(context, estimateRepository)
 
-        // Then add the burden estimates
-        val data = context.csvData<BurdenEstimate>()
-        return uploadBurdenEstimates(data, estimateRepository, context, path)
+        val request = context.request
+        if (request.raw().getAttribute("org.eclipse.jetty.multipartConfig") == null)
+        {
+            val multipartConfigElement = MultipartConfigElement(System.getProperty("java.io.tmpdir"))
+            request.raw().setAttribute("org.eclipse.jetty.multipartConfig", multipartConfigElement)
+        }
+
+        request.raw().getPart("file").inputStream.bufferedReader().use {
+
+            // Then add the burden estimates
+            val data = DataTableDeserializer.deserialize(it.readText(), BurdenEstimate::class, Serializer.instance).toList()
+            return saveBurdenEstimates(data, estimateRepository, context, path)
+        }
     }
 
-    private fun getValidResponsibilityPath(context: ActionContext, estimateRepository: BurdenEstimateRepository): ResponsibilityPath
+    open fun addBurdenEstimates(context: ActionContext, estimateRepository: BurdenEstimateRepository): String
```

Comment:
not introduced in this PR, but is there a reason for this to be an open function?

Reply:
I think for mocking in the tests.